Add hash command

sdcb opened the pull request:
This should fix https://github.com/Microsoft/vcpkg/issues/89.

Example: 
- vcpkg hash boost_1_62_0.tar.bz2
- vcpkg hash boost_1_62_0.tar.bz2 SHA1

sdcb committed:
Add hash file support.

msftclas commented:
Hi __@sdcb__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

alexkaratarakis commented:
Hello!
First of all, this looks good! We had previously implemented this with PR #98 , but decided to hold it off pending further comments on the subject. Also, #98 uses `certutil.exe` under the hood, because Get-FileHash is not available on Windows 7's version of powershell. 

About the default behavior of `vcpkg hash`, here are some options:
a) Default to SHA512, offer the option to specify hash function (this is what you have now)  
b) Default to showing all hashes that you are likely to need by default. SHA512 (needed for the ports) and SHA256, MD5 (etc?) that are usually given by websites to verify that what you downloaded is correct. Don't allow specifying hash.
c) Same as c, but allow specifying  hash.

What would you find more convenient?

sdcb commented:
I think option 1 would be more convenient because:
1. vcpkg itself uses SHA512 by default
2. I can copy hash data from console without worries about line-break

Commit:
Switch to using CertUtil to generate hash.

sdcb commented:
I switched to using CertUtil.exe instead of powershell Get-FileHash to make windows 7 works.
And fixed the localized issue that https://github.com/Microsoft/vcpkg/pull/98 have by search the second line.

alexkaratarakis commented:
Solid work!
Thanks 👍
